lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -81,32 +82,33 @@ def __init__(
         self.node_id_to_code: Dict[int, str] = {}
         self.imports: List[str] = []
 
-    def process_node(self, logical_node: Node, query_id: str, is_result_node: Optional[bool] = False) -> Any:
+    def process_node(
+        self, logical_node: Node, result: SycamoreQueryResult, is_result_node: Optional[bool] = False
+    ) -> Any:
         """Process the given node. Recursively processes dependencies first."""
 
+        query_id = result.query_id
         bind_contextvars(logical_node=logical_node)
-        # This is lifted up here to avoid serialization issues with Ray.
-
         if logical_node.node_id in self.processed:
             log.info("Already processed")
             return self.processed[logical_node.node_id]
         log.info("Executing dependencies")
         inputs: List[Any] = []
 
-        if self.trace_dir and not self.dry_run:
-            trace_dir = os.path.join(self.trace_dir, query_id, str(logical_node.node_id))
-            os.makedirs(trace_dir, exist_ok=True)
-        else:
-            trace_dir = None
-
         if self.cache_dir and not self.dry_run:
             cache_dir = os.path.join(self.cache_dir, logical_node.cache_key())
+            if result.execution is None:
```

Comment:
nit: can probably simplify this to not have if statements directly, e.g.
result.execution = result.execution or {}
result.execution[logical_node.node_id] = result.execution.get(logical_node.node_id, NodeExecution(...))

stylistic pref I suppose